src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkRepositoryFunction.java

```diff
@@ -323,6 +323,11 @@ private RepositoryDirectoryValue.Builder fetchInternal(
         markerData.put("FILE:" + entry.getKey(), entry.getValue());
       }
 
+      // Ditto for environment variables accessed via `getenv`.
+      for (String envKey : starlarkRepositoryContext.getAccumulatedEnvKeys()) {
+        markerData.put("LAZYENV:" + envKey, clientEnvironment.get(envKey));
```

Comment:
IMO we shouldn't need a separate "LAZYENV" class. We should just use the existing "ENV".

Reply:
Hm, but what about https://github.com/bazelbuild/bazel/blob/0ad0506594e76b2f2b01c805ce1a8af53e3c322a/src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryFunction.java#L371-L376 ? If I naively used `ENV:`, then we'd consistently invalidate whenever a rule implementation called `rctx.env("FOO")` without also predeclaring `"FOO"` in `repository_rule.environ`, right? Am I missing something? Or is that check somehow redundant and safe to remove? (For example, at first glance it seems that any modification to a .bzl file invalidates the repository rules contained in it. I haven't yet found the corresponding Bazel code to confirm that, though.)

Reply:
We should simply remove this check, IMO.

> For example, at first glance it seems that any modification to a .bzl file invalidates the repository rules contained in it. I haven't yet found the corresponding Bazel code to confirm that, though.

It's rather hidden: the transitive bzl digest is part of the "rule key" here https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java;drc=0ad0506594e76b2f2b01c805ce1a8af53e3c322a;l=570